### PR TITLE
Make Microbenchmarks Test Targets

### PR DESCRIPTION
**Pull request by ncteisen**
For #16850

### CODE REVIEW

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by jtattermusch** [comment]
tests failing:
```
I1015 13:43:59.186907959      18 ev_epoll1_linux.cc:116]     grpc epoll fd: 3
D1015 13:43:59.186959178      18 ev_posix.cc:169]            Using polling engine: epoll1
D1015 13:43:59.187067232      18 dns_resolver.cc:338]        Using native dns resolver
E1015 13:43:59.190355080      18 bm_cq_multiple_threads.cc:113] assertion failed: strcmp(grpc_get_poll_strategy_name(), "none") == 0 || strcmp(grpc_get_poll_strategy_name(), "bm_cq_multiple_threads") == 0
```

**Comment by ncteisen** [comment]
Does bazel have the ability to exclude certain pollers? 

For example. the fullstack benchmarks look like:

```
- name: bm_fullstack_streaming_ping_pong
  build: test
  language: c++
  headers:
  - test/cpp/microbenchmarks/fullstack_streaming_ping_pong.h
  src:
  - test/cpp/microbenchmarks/bm_fullstack_streaming_ping_pong.cc
  deps:
  - grpc_benchmark
  - benchmark
  - grpc++_test_util_unsecure
  - grpc_test_util_unsecure
  - grpc++_unsecure
  - grpc_unsecure
  - gpr_test_util
  - gpr
  - grpc++_test_config
  benchmark: true
  defaults: benchmark
  excluded_poll_engines:
  - poll
  - poll-cv
  platforms:
  - mac
  - linux
  - posix
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,965,242      Total (=)      1,965,242

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,909,029      Total (>)     10,909,024

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 1,965,242      Total (=)      1,965,242

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
10,909,023      Total (<)     10,909,027

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

File: test/cpp/microbenchmarks/BUILD [hunk 1]
```diff
@@ -134,41 +129,38 @@ grpc_cc_binary(
 
 grpc_cc_binary(
```
Comment (jtattermusch) [inline test/cpp/microbenchmarks/BUILD]: any reason why bm_fullstack_trickle is not added?

**Comment by jtattermusch** [comment]
@ncteisen friendly ping.   Are we going to add the tests I mentioned in https://github.com/grpc/grpc/pull/16882#pullrequestreview-171900250? Otherwise, we can merge.

**Comment by jtattermusch** [comment]
Also bm_call_create seems to have failed on ubsan: 

https://source.cloud.google.com/results/invocations/b17e5717-f082-460a-b5d4-1ece7b62490c/targets/%2F%2Ftest%2Fcpp%2Fmicrobenchmarks:bm_call_create/log

```
/usr/lib/gcc/x86_64-linux-gnu/4.9/../../../../include/c++/4.9/bits/stl_vector.h:780:16: runtime error: reference binding to null pointer of type 'const grpc_channel_filter *'
    #0 0x488fd7 in std::vector<grpc_channel_filter const*, std::allocator<grpc_channel_filter const*> >::operator[](unsigned long) /usr/lib/gcc/x86_64-linux-gnu/4.9/../../../../include/c++/4.9/bits/stl_vector.h:780:9
    #1 0x46cd6a in void BM_IsolatedFilter<Fixture<(grpc_channel_filter const*)0, 0u>, NoOp>(benchmark::State&) /proc/self/cwd/test/cpp/microbenchmarks/bm_call_create.cc:531:3
    #2 0x5b1301 in benchmark::internal::FunctionBenchmark::Run(benchmark::State&) /proc/self/cwd/external/com_github_google_benchmark/src/benchmark_register.cc:447:42
    #3 0x586532 in benchmark::internal::(anonymous namespace)::RunInThread(benchmark::internal::Benchmark::Instance const*, unsigned long, int, benchmark::internal::ThreadManager*) /proc/self/cwd/external/com_github_google_benchmark/src/benchmark.cc:166:17
    #4 0x584772 in benchmark::internal::(anonymous namespace)::RunBenchmark(benchmark::internal::Benchmark::Instance const&, std::vector<benchmark::BenchmarkReporter::Run, std::allocator<benchmark::BenchmarkReporter::Run> >*) /proc/self/cwd/external/com_github_google_benchmark/src/benchmark.cc:209:7
    #5 0x5821e1 in benchmark::internal::(anonymous namespace)::RunBenchmarks(std::vector<benchmark::internal::Benchmark::Instance, std::allocator<benchmark::internal::Benchmark::Instance> > const&, benchmark::BenchmarkReporter*, benchmark::BenchmarkReporter*) /proc/self/cwd/external/com_github_google_benchmark/src/benchmark.cc:430:11
    #6 0x5812ab in benchmark::RunSpecifiedBenchmarks(benchmark::BenchmarkReporter*, benchmark::BenchmarkReporter*) /proc/self/cwd/external/com_github_google_benchmark/src/benchmark.cc:544:5
    #7 0x5807e2 in benchmark::RunSpecifiedBenchmarks() /proc/self/cwd/external/com_github_google_benchmark/src/benchmark.cc:486:10
    #8 0x47ea7c in benchmark::RunTheBenchmarksNamespaced() /proc/self/cwd/test/cpp/microbenchmarks/bm_call_create.cc:832:37
    #9 0x47eac9 in main /proc/self/cwd/test/cpp/microbenchmarks/bm_call_create.cc:838:3
    #10 0x7fd033d2e82f in __libc_start_main (/lib/x86_64-linux-gnu/libc.so.6+0x2082f)
    #11 0x449b68 in _start (/b/f/w/bazel-out/k8-fastbuild/bin/test/cpp/microbenchmarks/bm_call_create.runfiles/com_github_grpc_grpc/test/cpp/microbenchmarks/bm_call_create+0x449b68)
```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,020,410      Total (=)      2,020,410

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,174,077      Total (>)     11,174,075

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,020,410      Total (=)      2,020,410

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,174,081      Total (=)     11,174,081

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Objective-C binary sizes
*****************STATIC******************
  New size                      Old size
 2,020,410      Total (=)      2,020,410

 No significant differences in binary sizes

***************FRAMEWORKS****************
  New size                      Old size
11,174,081      Total (>)     11,174,073

 No significant differences in binary sizes


```

**Comment by grpc-testing** [comment]
```
****************************************************************

libgrpc.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]


****************************************************************

libgrpc++.so

     VM SIZE        FILE SIZE
 ++++++++++++++  ++++++++++++++

  [ = ]       0        0  [ = ]



```

**Comment by grpc-testing** [comment]
```
[trickle] No significant performance differences
```

**Comment by grpc-testing** [comment]
```
Corrupt JSON data (indicates timeout or crash): 
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.new: 10
    bm_call_create.BM_IsolatedFilter_ClientChannelFilter_NoOp_.counters.old: 10


[microbenchmarks] No significant performance differences
```

**Comment by jtattermusch** [comment]
@ncteisen  please let me know once this is ready for review.

**Comment by ncteisen** [comment]
It is ready. All failures are unrelated. I have left TODOs for the remaining test targets

**Pull request merged**